### PR TITLE
Remove cruft, do some simple non-functional cleanup in the runtime

### PR DESCRIPTION
**Pull request by egernst**
As part of the effort of cleaning up the runtime code, I noticed:

packages which are not used in our code base
packages within virtcontainers which should be at top level
public functions which should be kept private
unused functions
Let's clean these up for better maintainability!

### CODE REVIEW

File: src/runtime/virtcontainers/network.go [hunk 1]
```diff
@@ -1486,7 +1486,7 @@ func addTxRateLimiter(endpoint Endpoint, maxRate uint64) error {
 	var netPair *NetworkInterfacePair
 	var linkName string
 	switch ep := endpoint.(type) {
-	case *VethEndpoint, *IPVlanEndpoint, *TuntapEndpoint, *BridgedMacvlanEndpoint:
+	case *VethEndpoint, *IPVlanEndpoint, *TuntapEndpoint, *MacvlanEndpoint:
```
Comment (mcastelino) [inline src/runtime/virtcontainers/network.go:1489]: @egernst ack. The interconnect model dictates how we connect. It should be macvlan and not bridgedmacvlan

**Review by sameo — APPROVED** [review]

**Comment by fidencio** [comment]
```
main_test.go:24: File is not `gofmt`-ed with `-s` (gofmt)
	vc "github.com/kata-containers/kata-containers/src/runtime/virtcontainers"
make: *** [Makefile:40: static-checks] Error 1
```

**Review by fidencio — APPROVED** [review]
Please, fix the error found by the static checks and this one will be good to be merged.
@egernst, thanks!

**Comment by egernst** [comment]
Pushed one more time after running gofmt on the changed files.

**Pull request merged**